src/workerd/io/trace.h

```diff
@@ -477,4 +287,158 @@ inline kj::String truncateScriptId(kj::StringPtr id) {
   return kj::str(truncatedId);
 }
 
+// =======================================================================================
+// Span tracing
+//
+// TODO(cleanup): As of now, this aspect of tracing is actually not related to the rest of this
```

Comment:
It's occurring to me that this stuff really belongs in observer.h. You may have noticed I renamed `SpanCollector` to `SpanObserver`, even. And, currently the way that `IoContext` finds its parent span is via `RequestObserver::getSpan()`. The `Tracer` object passed down through `WorkerEntrypoint` seems to be entirely redundant.

Reply:
Opened #142 with the simple cleanup here.